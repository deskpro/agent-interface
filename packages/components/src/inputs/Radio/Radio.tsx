import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-forms.css";

export interface RadioProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  label: string;
}

const Radio: React.SFC<RadioProps> = ({ className, id, label, ...props }) => (
  <div className="dp-RadioItem">
    <input
      type="radio"
      id={id}
      className={classNames("dp-Input Input--radio", className)}
      {...props}
    />
    <label htmlFor={id}>{label}</label>
  </div>
);

export default Radio;
