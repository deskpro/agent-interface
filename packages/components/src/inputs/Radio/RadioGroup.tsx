import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-forms.css";

const RadioGroup: React.FC<React.HTMLAttributes<HTMLElement>> = ({
  children,
  className,
  ...props
}) => (
  <div className={classNames("dp-RadioWrapper", className)} {...props}>
    {children}
  </div>
);

export default RadioGroup;
