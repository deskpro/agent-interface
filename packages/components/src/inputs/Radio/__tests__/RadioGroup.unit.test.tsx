import * as React from "react";
import { render } from "react-testing-library";

import Radio from "../Radio";
import RadioGroup from "../RadioGroup";

describe("<Radio />", () => {
  it("should match its snapshot", () => {
    const handleChange = jest.fn();
    const { container } = render(
      <RadioGroup className="radio-group-test">
        <Radio
          id="one"
          label="Option one"
          value="one"
          onChange={handleChange}
        />
        <Radio
          id="two"
          label="Option two"
          value="two"
          onChange={handleChange}
        />
        <Radio
          id="three"
          label="Option three"
          value="three"
          onChange={handleChange}
        />
      </RadioGroup>
    );
    expect(container.querySelector(".radio-group-test")).toMatchSnapshot();
  });
});
