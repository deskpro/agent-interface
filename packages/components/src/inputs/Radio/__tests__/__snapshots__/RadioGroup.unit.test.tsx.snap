// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> it should match its snapshot 1`] = `
<div
  class="dp-RadioWrapper radio-group-test"
>
  <div
    class="dp-RadioItem"
  >
    <input
      class="dp-Input Input--radio"
      id="one"
      type="radio"
      value="one"
    />
    <label
      for="one"
    >
      Option one
    </label>
  </div>
  <div
    class="dp-RadioItem"
  >
    <input
      class="dp-Input Input--radio"
      id="two"
      type="radio"
      value="two"
    />
    <label
      for="two"
    >
      Option two
    </label>
  </div>
  <div
    class="dp-RadioItem"
  >
    <input
      class="dp-Input Input--radio"
      id="three"
      type="radio"
      value="three"
    />
    <label
      for="three"
    >
      Option three
    </label>
  </div>
</div>
`;
